### PR TITLE
Codechange: Realign SDL driver with SDL2 driver to ease maintenance.

### PR DESCRIPTION
**Pull request by techgeeknz**
Renamed _sdl_screen, reformatted sourcecode, and updated comments so as to allow the two versions of the SDL video driver to be more easily compared and maintained.

### CODE REVIEW

File: src/video/sdl_v.cpp [hunk 1]
```diff
@@ -155,17 +155,22 @@ static void DrawSurfaceToScreen()
...
+			SDL_BlitSurface(_sdl_surface, nullptr, _sdl_realscreen, nullptr);
 		}
+
 		SDL_UpdateRect(_sdl_realscreen, 0, 0, 0, 0);
 	} else {
-		if (_sdl_screen != _sdl_realscreen) {
+		if (_sdl_surface != _sdl_realscreen) {
 			for (int i = 0; i < n; i++) {
-				SDL_BlitSurface(_sdl_screen, &_dirty_rects[i], _sdl_realscreen, &_dirty_rects[i]);
+				SDL_BlitSurface(
```
Comment (LordAro) [inline src/video/sdl_v.cpp]: I'd change the line in SDL2, tbh. OTTD doesn't have line length limits, and this line isn't even that long

Comment (techgeeknz) [reply src/video/sdl_v.cpp]: Good idea; I am more than willing to do that. How do I edit a pull request?

Comment (techgeeknz) [reply src/video/sdl_v.cpp]: Never mind, I figured it out ;)

**Review by LordAro — APPROVED** [review]
I'm not sure what the long term plan for the SDL1 driver is, it might be going away at some point. But in the meantime, this isn't a bad thing. If it's sticking around though, would be nice to reduce the code duplication further (using member variables to get rid of globals, etc)

**Comment by andythenorth** [comment]
Thanks for the contribution 😄

**Comment by techgeeknz** [comment]
Once I get my head around OpenTTD's input, video, and windowing systems, I am itching to squash #7006 and other GUI responsiveness issues by redesigning the video and windowing systems to impose hard limits on update times. I don't see any reason why the mouse pointer shouldn't update at least once every 30 ms; and, ideally, the windowing system should feel like it's glued to the mouse button :)

No, I don't expect it to be easy.

**Pull request merged**